### PR TITLE
Fix automatic order cancellation

### PR DESCRIPTION
**Pull request by anastasiabalaban**
The order is not cancelled if the payment is cancelled and automatic cancellation option is enabled.

Changes:
- also cancel order when the `onCancel` callback function is called
- use the correct `removeOrderPaymentHashes` function name
- `cancel_order` is deprecated, used `update_status` instead

### CODE REVIEW

File: includes/paypro/wc/woocommerce.php [hunk 1]
```diff
@@ -66,11 +66,11 @@ public function cancelOrder(WC_Order $order, $payment_hash)
         $message = sprintf(__('PayPro payment cancelled (%s)', 'paypro-gateways-woocommerce'), $payment_hash);
 
         if(PayPro_WC_Plugin::$settings->automaticCancellation())
-            $order->cancel_order($message);
+            $order->update_status('cancelled', $message);
         else
             $order->add_order_note($message);
 
-        $this->removeOrderPaymentHash($order);
+        $this->removeOrderPaymentHashes($order);
```
Comment (henkjanwils) [inline includes/paypro/wc/woocommerce.php:73]: Was this a bug too?

Comment (anastasiabalaban) [reply includes/paypro/wc/woocommerce.php:73]: Yes, the error was thrown after the order cancelling.

**Comment by henkjanwils** [comment]
There's still a "version 2.0.0"
Screenshot 2024-04-10 at 14 09 15
Screenshot 2024-04-10 at 14 08 56

**Comment by henkjanwils** [comment]
We should add a "new version" here too.
Screenshot 2024-04-10 at 14 11 55

**Review by henkjanwils — APPROVED** [review]

**Pull request merged**